oslabs-beta/Kubiquity
# Add Pod names, Fix query

WHAT DOES THIS PR DO?

- Add names instead of using 0, 1,...,etc for podId
- Now retrieves list of all running pods
- memory now in bytes

HOW TO TEST THIS PR:

1. Ensure minikube is running with `minikube start`
2. Run `gh pr checkout 32 && npm install && npm run backend` or  `gh pr checkout 32 && npm install && npm run dev` (must change from mock data to live data in App.jsx)
3. If you ran `npm run backend`, send a GET request to `localhost:3000/metrics` and ensure you get an array of objects back with podId and memory with no memory values of 0 and podIds of undefined.
4. If you ran `npm run dev` ensure the app loads the new metrics.

ADDITIONAL NOTES:

- Previously we were misunderstanding what we were receiving where one element was an ID of some sort and the second parameter was the sum of all memory usage by the Pods. That is why we only sent an array of two objects. Now, refining our queries, we now retrieve every running POD that is running.
- Note to frontend: We might have to change from MB to just bytes as some of the memory metrics of some pods are way too small to convert to megabytes.
- Integration with electron still seems to be a little iffy as metrics don't show up on the app. Definitely need a little help in that area.
